SCAutolib/utils.py

```diff
@@ -0,0 +1,104 @@
+from cryptography.hazmat.primitives import serialization
+from cryptography.hazmat.primitives.asymmetric import rsa
+from pathlib import Path
+
+from enum import Enum
+
+from SCAutolib import run, logger, TEMPLATES_DIR
+
+
+class OSVersion(Enum):
+    """
+    Enumeration for Linux versions. Used for more convenient checks.
+    """
+    Fedora = 1
+    RHEL_9 = 2
+    RHEL_8 = 3
+
+
+def _check_selinux():
+    """
+    Checks if specific SELinux module for virtual smart card is installed.
+    This is implemented be checking the hardcoded name for the module
+    (virtcacard) to be present in the list of SELinux modules. If this name is
+    not present in the list, then virtcacard.cil file would be created in conf
+    or subdirectory in the CA directory specified by the configuration file.
+    """
+    result = run("semodule -l", print_=False)
+    if "virtcacard" not in result.stdout:
+        logger.debug(
+            "SELinux module for virtual smart cards is not present in the "
+            "system. Installing...")
+
+        run(["semodule", "-i", f"{TEMPLATES_DIR}/virtcacard.cil"])
+
+        run(["systemctl", "restart", "pcscd"])
```

Comment:
To discuss: Should we restart services from one place (controller) or from diverse methods and functions? IMHO, this should be controlled by controller.

Reply:
Needs more discussion